uaclient/entitlements/tests/conftest.py

```diff
@@ -3,51 +3,68 @@
 from uaclient import config
 
 try:
-    from typing import Any, Dict, List  # noqa
+    from typing import Any, Dict, List, Optional  # noqa
 except ImportError:
     # typing isn't available on trusty, so ignore its absence
     pass
 
 
-def machine_token(entitlement_type: str) -> "Dict[str, Any]":
+def machine_token(
+    type: str,
+    *,
+    affordances: "Dict[str, Any]" = None,
+    directives: "Dict[str, Any]" = None,
+    entitled: "Optional[bool]" = True,
+    obligations: "Dict[str, Any]" = None,
+    suites: "List[str]" = None
+) -> "Dict[str, Any]":
     return {
+        "resourceTokens": [{"Type": type, "Token": "%s-token" % type}],
         "machineToken": "blah",
         "machineTokenInfo": {
             "contractInfo": {
                 "resourceEntitlements": [
-                    {"type": entitlement_type, "entitled": True}
+                    machine_access(
+                        type,
+                        affordances=affordances,
+                        directives=directives,
+                        entitled=entitled,
+                        obligations=obligations,
+                        suites=suites,
+                    )
                 ]
             }
         },
     }
 
 
 def machine_access(
-    entitlement_type: str,
+    type: str,
     *,
     affordances: "Dict[str, Any]" = None,
     directives: "Dict[str, Any]" = None,
+    entitled: "Optional[bool]" = True,
+    obligations: "Dict[str, Any]" = None,
     suites: "List[str]" = None
 ) -> "Dict[str, Any]":
     if affordances is None:
         affordances = {"series": []}  # Will match all series
     if suites is None:
         suites = ["xenial"]
+    if obligations is None:
+        obligations = {"enableByDefault": True}
     if directives is None:
         directives = {
-            "aptURL": "http://{}".format(entitlement_type.upper()),
+            "aptURL": "http://{}".format(type.upper()),
             "aptKey": "APTKEY",
             "suites": suites,
         }
     return {
-        "resourceToken": "TOKEN",
-        "entitlement": {
-            "obligations": {"enableByDefault": True},
-            "type": entitlement_type,
-            "entitled": True,
-            "directives": directives,
-            "affordances": affordances,
-        },
+        "obligations": obligations,
+        "type": type,
+        "entitled": True,
```

Comment:
Should this be using the value of `entitled`?

Reply:
yes, nice one